rpcs3/Emu/RSX/Overlays/overlay_controls.cpp

```diff
@@ -397,30 +397,48 @@ namespace rsx
 							continue;
 						default:
 							ctr += 4;
+							break;
 						}
 					}
 
 					lines.emplace_back(line_begin, ctr);
-					const auto max_region_w = std::max<f32>(text_extents_w, w);
+					const f32 max_region_w = std::max<f32>(text_extents_w, w);
 					const f32 offset_extent = (alignment == text_align::center ? 0.5f : 1.0f);
+					const f32 size_px = renderer->get_size_px() * 0.5f;
 
-					for (auto p : lines)
+					for (const auto& [begin, end] : lines)
 					{
-						if (p.first >= p.second)
+						if (begin >= end)
 							continue;
 
-						const f32 line_length = result[p.second - 1].values[0] - result[p.first].values[0];
-						const bool wrapped = std::fabs(result[p.second - 1].values[1] - result[p.first + 3].values[1]) >= (renderer->get_size_px() * 0.5f);
-
-						if (wrapped)
-							continue;
-
-						if (line_length < max_region_w)
+						// Move all glyphs by the correct amount to get a nice alignment.
+						for (u32 i_begin = begin, i_next = begin + 4;; i_next += 4)
```

Comment:
This loop can be avoided. It is only needed if wrapping is required, otherwise it should not be run.

Reply:
refactored it a bit.
It now prefers the simple path if possible